test/params_test.c

```diff
@@ -465,7 +458,7 @@ static int test_case_variant(OSSL_PARAM *params, const struct provider_dispatch_
         || !TEST_str_eq(app_p5, p5_init)        /* "provider" value */
         || !TEST_ptr(p = OSSL_PARAM_locate(params, "p6"))
         || !TEST_size_t_eq(p->return_size, sizeof(p6_init)) /* "provider" value */
-        || !TEST_str_eq(app_p6, p6_init)        /* "provider" value */
+        || !TEST_str_eq(p->data, p6_init)        /* "provider" value */
```

Comment:
Restore

Reply:
This doesn't work.  The `app_p6` buffer isn't changed, the pointer that "p6" points too does.  `app_p6` will always have the application value.